### PR TITLE
[apex] Flagging of System.debug

### PR DESCRIPTION
**Pull request by sgorbaty**
Hey @jsotuyod,

This rule flags when potentially sensitive variables are in System.debug(), which would leak the data to the end user. 

Cheers,
Sergey

### CODE REVIEW

**Comment by jsotuyod** [comment]
@sgorbaty thanks again! I'll try to look into this ASAP.

File: pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java [hunk 1]
```diff
@@ -35,14 +50,49 @@ public Object visit(ASTUserClass node, Object data) {
             return data;
         }
 
+        collectBenignVariables(node);
```
Comment (jsotuyod) [inline pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java]: be careful here. A single rule instance will be used to analyze multiple files. That means the `whiteListedVariables` will accumulate all names from previous files and could produce false negatives. You should either clear the set for each class, or move it to a local variable.

Comment (sgorbaty) [inline pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java]: Right, I address that with Helper.getFQVariableName that includes class name.

Comment (sgorbaty) [inline pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java]: I learned that one quickly :)

Comment (jsotuyod) [inline pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java]: In that case, you are not generating False Positives, but are preventing stale Strings from being garbage collected. I'd still recommend clearing it as a final step when done visiting the `ASTUserClass`.

Comment (sgorbaty) [inline pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java]: Sure, will do.

**Comment by jsotuyod** [comment]
Merged! Thanks again.

**Pull request merged**